src/main/java/org/b612foundation/adam/opm/OpmWriter.java

```diff
@@ -0,0 +1,109 @@
+package org.b612foundation.adam.opm;
+
+import javax.swing.plaf.nimbus.State;
+
+/**
+ * Writer class that turns OrbitParameterMessage objects into string representations
+ */
+public class OpmWriter {
+
+  public static String toCcsdsOpmString(OrbitParameterMessage opm) {
+    StringBuilder builder = new StringBuilder();
+
+    OdmCommonHeader header = opm.getHeader();
+    builder.append("CCSDS_OPM_VERS = " + opm.getCcsds_opm_vers() + "\n");
+    builder.append("CREATION_DATE = " + header.getCreation_date() + "\n");
+    builder.append("ORIGINATOR = " + header.getOriginator() + "\n");
+
+    OdmCommonMetadata metadata = opm.getMetadata();
```

Comment:
Paranoia here and everywhere: might want to check for null and maybe throw something like IllegalArgument instead?

Reply:
I added checks here and in other places to make it more robust and informative.